### PR TITLE
#66: ColumnName now uses table_like_name

### LINKED ISSUE #66: ColumnName uses TableName instead of TableLikeName
# Background

- Columns can be used with tables and views and for this they need to accept TableLikeNames

### PR DESCRIPTION
**Pull request by MarleneKress79789**
I also changed the names of functions such as ColumnName.table_name -> ColumnName.table_like_name for consistency sake. I am not sure if this is good or if we rather leave it for better consistency between versions.

Fixes #66

### CODE REVIEW

File: .github/workflows/language_container.yaml [hunk 1]
```diff
@@ -13,6 +13,10 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - uses: abatilo/actions-poetry@v2.1.4
         with:
-          poetry-version: 1.2.0
+          poetry-version: 1.2.2
+      - name: uninstall
+        run: pip uninstall -y urllib3 chardet
```
Comment (Nicoretti) [inline .github/workflows/language_container.yaml:18]: why is this needed?

Comment (MarleneKress79789) [reply .github/workflows/language_container.yaml:18]: there is a problem with versions in urllib chardet and request that only pops up in the github actions , installing via poetry locally works fine. 

`
Run ./build_language_container.sh
/usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.12) or chardet (3.0.4) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "

No module named 'poetry.core.constraints'`

this is the only solution i found that worked. the pinning the versions in pyproject.toml did not solve it for me. it also might be related to the language container somehow. but torsten also did not know another solution when he had a quick look. if you have a better idea let me know.

Comment (Nicoretti) [reply .github/workflows/language_container.yaml:18]: Hi @MarleneKress79789 ,

I think the dependency issue was caused by the unpinned `python` version in the CI build,[see also PR-69](https://github.com/exasol/data-science-utils-python/pull/69/files).

**Review by Nicoretti — APPROVED** [review]
